crypto/rand/rand_win.c

```diff
@@ -162,7 +162,7 @@ int rand_pool_add_additional_data(RAND_POOL *pool)
     return rand_pool_add(pool, (unsigned char *)&data, sizeof(data), 0);
 }
 
-# if !OPENSSL_API_1_1_0
+# if !OPENSSL_API_1_1_0 || defined(FIPS_MODE)
```

Comment:
Are you sure to enable it in FIPS mode? Or did you mean
```
# if !OPENSSL_API_1_1_0 && !defined(FIPS_MODE)
```

Reply:
Yes, I did mean that!

Reply:
Hmmm... but why? The code is marked deprecated-in-1.1.1 and not called anywhere.

```
msp@msppc:~/src/openssl$ git grep -E -nH 'RAND_(event|screen)' pr-9035
pr-9035:crypto/rand/rand_win.c:166:int RAND_event(UINT iMsg, WPARAM wParam, LPARAM lParam)
pr-9035:crypto/rand/rand_win.c:172:void RAND_screen(void)
pr-9035:doc/man3/RAND_add.pod:5:RAND_add, RAND_poll, RAND_seed, RAND_status, RAND_event, RAND_screen,
pr-9035:doc/man3/RAND_add.pod:25: int RAND_event(UINT iMsg, WPARAM wParam, LPARAM lParam);
pr-9035:doc/man3/RAND_add.pod:26: void RAND_screen(void);
pr-9035:doc/man3/RAND_add.pod:74:RAND_event() and RAND_screen() are equivalent to RAND_poll() and exist
pr-9035:doc/man3/RAND_add.pod:84:RAND_event() returns RAND_status().
pr-9035:doc/man3/RAND_add.pod:98:RAND_event() and RAND_screen() were deprecated in OpenSSL 1.1.0 and should
pr-9035:include/openssl/rand.h:68:DEPRECATEDIN_1_1_0(void RAND_screen(void))
pr-9035:include/openssl/rand.h:69:DEPRECATEDIN_1_1_0(int RAND_event(UINT, WPARAM, LPARAM))
pr-9035:util/libcrypto.num:1331:RAND_event                              1318    3_0_0   EXIST:_WIN32:FUNCTION:DEPRECATEDIN_1_1_0
pr-9035:util/libcrypto.num:1816:RAND_screen                             1804    3_0_0   EXIST:_WIN32:FUNCTION:DEPRECATEDIN_1_1_0
```

Reply:
Is it because it is still referenced in libcrypto.num?

```
pr-9035:util/libcrypto.num:1331:RAND_event                              1318    3_0_0   EXIST:_WIN32:FUNCTION:DEPRECATEDIN_1_1_0
pr-9035:util/libcrypto.num:1816:RAND_screen                             1804    3_0_0   EXIST:_WIN32:FUNCTION:DEPRECATEDIN_1_1_0
```

Reply:
Sorry, I wasn't clear. What *you* wrote is what I meant! I've fixed it now.

Reply:
But the FIPS provider does not reuse libcrypto.num, right?

Reply:
> Sorry, I wasn't clear. What _you_ wrote is what I meant! I've fixed it now.

Ok, now the world is in order again :-)

Reply:
> But the FIPS provider does not reuse libcrypto.num, right?

Right. It has its own num file which exports exactly one function:

https://github.com/openssl/openssl/blob/master/util/providers.num